### ISSUE TITLE
Autocompletion and function help not working in FreeBSD with Pylance

### ISSUE THREAD

**Issue by Huskers2020** [microsoft/vscode-jupyter#7459]
## Environment data

-   VS Code version: 1.56
-   Jupyter Extension version (available under the Extensions sidebar): 2021.6.832593372
-   Python Extension version (available under the Extensions sidebar): 2021.5.926500501
-   OS (Windows | Mac | Linux distro) and version: FreeBSD
-   Python and/or Anaconda version: 3.8
-   Type of virtual environment used: N/A
-   Jupyter server running:  -NA

I'm not sure if this extension supports FreeBSD; I believe it should work if my understanding is incorrect. My apologies.

I have also provided steps on how to get JediLSP working for anyone interested; I cannot comment if this was working before with Jupyter as JediLS does work flawlessly with Python on FreeBSD.

## Actual behavior

1 - Pylance LS does not recognize FreeBSD as an OS platform and fails on startup

**[Error: No native build was found for paltform=freebsd arch=x64 runtime=electron abi=87 uv=1 libc=glibc ..**

2- On setting to JediLSP, the LS starts and fails with error:- 

**[Error: command 'python.analysis.restartLanguageServer' already exists.**

There was an issue logged last year and was addressed, but this issue still prevails on FreeBSD - [13441](https://github.com/microsoft/vscode-python/issues/13441) & [13474](https://github.com/microsoft/vscode-python/issues/13474)

**Workaround to make JediLSP work:-**
To make JediLSP work set the language server to "JediLSP" and open "JSON" settings in VSCode along with a python code on the side. 
Close VSCode
Open VSCode and let all the initialization of Jupyter and Language Server complete. (Review Jupyter output logs to check there are no errors except for the first error on the platform.)
Open an existing Jupyter notebook and watch the Jupyter output log; there should be no errors.
Move your mouse over Jupiter notebook cell python code and watch help info pop up.
In a new cell, type python code and watch autocompletion work.

For the above to work, I had commented out the following in my JSON settings -

**"jupyter.experiments.optOutFrom": ["NativeNotebookEditor"],**

Note:- I did notice at times there was lag; restarting VSCode resolved the issue.

## Steps to reproduce:

[**NOTE**: Self-contained, minimal reproducing code samples are **extremely** helpful and will expedite addressing your issue]

1. Install Jupyter extension(It automatically pulls in Python and Pylance)
2. open settings and set the language server to "Pylance" and for Jedi, set it to "JediLSP"
3. Close and open VSCode and check output log under Jupyter and Langauge Server.

</details>

**Comment by jakebailey** [comment]
Apologies for the double transfer, clicked the wrong `vscode-` repo. 🙂

**Comment by rchiodo** [comment]
The jupyter errors are benign. It means there's no ZMQ support for FreeBSD. We'll still run without it. Anything else seems to already be covered by what Jake said.

**Comment by Huskers2020** [comment]
> Where did you see the message?
> 
> ```
> [Error: No native build was found for paltform=freebsd arch=x64 runtime=electron abi=87 uv=1 libc=glibc ..
> ```
> 
> There shouldn't be any reason why we don't work on FreeBSD, other than that typeshed doesn't recognize or test anything other than the major three platforms. The only compiled code we have is for IntellICode, which should gracefully fail to load, and can be tested pretty easily by not having IC installed.
> 
> "command 'python.analysis.restartLanguageServer' already exists." is from the Python extension, and is more an artifact of how language servers work there; you have to reload the editor when you change LSs (though this particular message should not appear).

#

I did restart the IDE and the IDE also pop up a notification  at lower right corner suggesting "reload"

**Comment by Huskers2020** [comment]
s

> As far as I can tell, there are no official builds of VS Code for FreeBSD, and Pylance does not support unofficial builds and won't run on one. You should be using JediLSP instead, though the message about the command existing is likely an artifact of some code in the Python extension that only knows about Pylance getting disabled too late. Those are likely covered by the two issues you linked in the original issue: [microsoft/vscode-python#13474](https://github.com/microsoft/vscode-python/issues/13474) [microsoft/vscode-python#13441](https://github.com/microsoft/vscode-python/issues/13441)
> 
> The rest all appears to be coming from the Jupyter extension and happens regardless of the language server choice.

The issue I notice is JediLSP by itself does not work in the Jupyter notebook and for auto-completion to work Pylance has to be installed. Post-installation one has to follow the steps that I have outlined above to make it work. This is a nuance one has to go through every time IDE is launched.

The same JediLSP works on python code flawlessly.

**Comment by Huskers2020** [comment]
> The jupyter errors are benign. It means there's no ZMQ support for FreeBSD. We'll still run without it. Anything else seems to be already covered by what Jake said.

ZMQ libs are available, and I noticed it got installed.

After a little research, I could gather the issue is with the installation of Pylance extension. A little digging under output logs (Python), I now see a message buried in indicating "ms-python.vscode-pylance not installed" though the IDE shows it as installed.

Pylance and Python Language Server logs are empty when set to Pylance.

So even though the IDE shows properties related to Pylance in IDE, indicating the installation was successful and complete but in reality, it is not complete. I tried to install it manually, and it failed due to version differences(refer to my note above). Not sure how the IDE is installing it and marking the installation as complete and successful.

I did check Pylance on Arch Linux with ver 1.56, and it did work (before upgrading to 1.60). On FreeBSD, 1.60 is yet to be ported, so I can only validate it against 1.56. 

If there is the possibility to force the installation of this extension manually on FreeBSD, do let me know, and I will give it a try.

For now, I highly suspect the installation is the issue.
